### PR TITLE
add maps api to website clean pull request

### PR DESCRIPTION
**Pull request by ntarn**
Please ignore walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java  for now, I have been able to make a new branch with just the week5 commits, but was unable to remove this commit because it was underneath a lot of other commits, sorry about that.

### CODE REVIEW

File: portfolio/src/main/webapp/js/mapscript.js [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+    'I aim to use data science to help with wildlife conservation and environmental issues. '+
+    'Currently, I am working on a natural language processing project that analyzes newspaper topics for the Duke Chronicle. '+
+    'Additionally, I am conducting research on identifying energy infrastructures using computer vision with Kyle Bradbury.'+
+    '</p>'+
+    '</div>'+
+    '</div>';
+  const sunnyvaleLandmark = [map, 37.402601, -122.015315, 'Sunnyvale', 'Where I\'m based'];
+  const dukeLandmark = [map, 36.001339, -78.938218, 'Duke University', contentString]
+  addLandmark(sunnyvaleLandmark[0], sunnyvaleLandmark[1], sunnyvaleLandmark[2], sunnyvaleLandmark[3], sunnyvaleLandmark[4]);
+  addLandmark(dukeLandmark[0], dukeLandmark[1], dukeLandmark[2], dukeLandmark[3], dukeLandmark[4]);
```
Comment (ntarn) [inline portfolio/src/main/webapp/js/mapscript.js:222]: I added the constants here bc I don't expect all of these values to ever change.

File: portfolio/src/main/webapp/js/mapscript.js [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+    '<div id="bodyContent">'+
+    '<p><b>Major: Computer Science and Biology</b><br>' +
+    'I aim to use data science to help with wildlife conservation and environmental issues. '+
+    'Currently, I am working on a natural language processing project that analyzes newspaper topics for the Duke Chronicle. '+
+    'Additionally, I am conducting research on identifying energy infrastructures using computer vision with Kyle Bradbury.'+
+    '</p>'+
+    '</div>'+
+    '</div>';
+  const sunnyvaleLandmark = [map, 37.402601, -122.015315, 'Sunnyvale', 'Where I\'m based'];
+  const dukeLandmark = [map, 36.001339, -78.938218, 'Duke University', contentString]
```
[the remark this answers is not in the document]
Comment (ntarn) [reply portfolio/src/main/webapp/js/mapscript.js:220]: changed to what we discussed in 1:1

File: portfolio/src/main/webapp/js/mapscript.js [hunk 3]
```diff
@@ -0,0 +1,240 @@
+/** Creates a map with dark grayscale styling and adds marker information for main locations. */
+function createMap() {
+  // Create a new StyledMapType object, passing it an array of styles, and the name to be displayed on the map type control.
```
Comment (hayleejane3) [inline portfolio/src/main/webapp/js/mapscript.js:3]:
```suggestion
  // Create a new {@code StyledMapType} object,  with an array of styles, and the name to be displayed on the map type control.
```

File: portfolio/src/main/webapp/js/mapscript.js [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+  map.mapTypes.set('styled_map', styledMapType);
+  map.setMapTypeId('styled_map');
+  
+  const sunnyvaleLat = 37.402601;
+  const sunnyvaleLong = -122.015315;
+  const sunnyvaleTitle = 'Sunnyvale';
+  const sunnyvaleDes = 'Where I\'m based';
+  const dukeLat = 36.001339;
+  const dukeLong = -78.938218;
+  const dukeTitle = 'Duke University';
```
Comment (hayleejane3) [inline portfolio/src/main/webapp/js/mapscript.js:214]: Create these as object-type constants instead of individual ones for each part of each location.

File: walkthroughs/week-5-tdd/intro/src/test/java/com/google/sps/GreeterTest.java [hunk 5]
```diff
@@ -30,4 +30,14 @@ public void testGreeting() {
 
     Assert.assertEquals("Hello Ada", greeting);
   }
+  
+  @Test
```
Comment (hayleejane3) [inline walkthroughs/week-5-tdd/intro/src/test/java/com/google/sps/GreeterTest.java:34]: These changes should. not be part of this PR.

Comment (ntarn) [reply walkthroughs/week-5-tdd/intro/src/test/java/com/google/sps/GreeterTest.java:34]: I saw that as well, and will do so from now on! More details about how I'm committing to making my branches clean in our pm's :)

File: portfolio/src/main/webapp/js/mapscript.js [hunk 6]
```diff
@@ -1,6 +1,16 @@
...
+    this.lng = lng;
+    this.title = title;
+    this.des = des;
+  }
+}
+
 /** Creates a map with dark grayscale styling and adds marker information for main locations. */
 function createMap() {
-  // Create a new StyledMapType object, passing it an array of styles, and the name to be displayed on the map type control.
+  // Create a new {@code StyledMapType} object,  with an array of styles, and the name to be displayed on the map type control.
```
Comment (hayleejane3) [inline portfolio/src/main/webapp/js/mapscript.js:13]:
```suggestion
  // Create a new {@code StyledMapType} object, with an array of styles, and the name to be displayed on the map type control.
```

File: portfolio/src/main/webapp/js/mapscript.js [hunk 7]
```diff
@@ -205,13 +215,7 @@ function createMap() {
...
   map.setMapTypeId('styled_map');
   
-  const sunnyvaleLat = 37.402601;
-  const sunnyvaleLong = -122.015315;
-  const sunnyvaleTitle = 'Sunnyvale';
-  const sunnyvaleDes = 'Where I\'m based';
-  const dukeLat = 36.001339;
-  const dukeLong = -78.938218;
-  const dukeTitle = 'Duke University';
+  sunnyvale = new Landmark(37.402601, -122.015315, 'Sunnyvale', 'Where I\'m based');
```
Comment (hayleejane3) [inline portfolio/src/main/webapp/js/mapscript.js:218]: make this a const and use CAPS in the naming

File: portfolio/src/main/webapp/js/mapscript.js [hunk 8]
```diff
@@ -205,13 +215,7 @@ function createMap() {
...
   const dukeDes = '<div id="content">'+
```
Comment (hayleejane3) [inline portfolio/src/main/webapp/js/mapscript.js:219]: Full words for the names please.

File: portfolio/src/main/webapp/js/mapscript.js [hunk 9]
```diff
@@ -224,8 +228,9 @@ function createMap() {
     '</p>'+
     '</div>'+
     '</div>';
-  addLandmark(map, sunnyvaleLat, sunnyvaleLong, sunnyvaleTitle, sunnyvaleDes);
-  addLandmark(map, dukeLat, dukeLong, dukeTitle, dukeDes);
+  duke = new Landmark(36.001339, -78.938218, 'Duke University', dukeDes);
```
Comment (hayleejane3) [inline portfolio/src/main/webapp/js/mapscript.js:231]: const here too

File: portfolio/src/main/webapp/js/mapscript.js [hunk 10]
```diff
@@ -1,6 +1,16 @@
+/** A class that helps with organizing landmark information. */
+class Landmark {
+  constructor(lat, lng, title, des) {
+    this.lat = lat;
+    this.lng = lng;
+    this.title = title;
+    this.des = des;
```
Comment (hayleejane3) [inline portfolio/src/main/webapp/js/mapscript.js:7]: "des" is not very obvious in meaning. Rename to make it clearer.

File: portfolio/src/main/webapp/js/mapscript.js [hunk 11]
```diff
@@ -215,8 +215,8 @@ function createMap() {
   map.mapTypes.set('styled_map', styledMapType);
   map.setMapTypeId('styled_map');
   
-  sunnyvale = new Landmark(37.402601, -122.015315, 'Sunnyvale', 'Where I\'m based');
-  const dukeDes = '<div id="content">'+
+  const SUNNYVALE = new Landmark(37.402601, -122.015315, 'Sunnyvale', 'Where I\'m based');
```
Comment (hayleejane3) [inline portfolio/src/main/webapp/js/mapscript.js:218]: SUNNYVALE_LANDMARK (same for Duke)

File: portfolio/src/main/webapp/js/mapscript.js [hunk 12]
```diff
@@ -215,8 +215,8 @@ function createMap() {
...
+  const DUKEDESCRIPTION = '<div id="content">'+
```
Comment (hayleejane3) [inline portfolio/src/main/webapp/js/mapscript.js:219]: Use underscores to separate words in Const names.
